addon/titleUrlPage.js

```diff
@@ -3,20 +3,20 @@ browser.runtime.onMessage.addListener(updateUrl)
 updateUrl()
 
 function updateUrl() {
-    var title = getTitle();
-    if(title){
-       document.title = title;
-    }
-}
+	var url = document.URL;
+
+	url = url.replace(/(https?:\/\/[^\/]*\/)[^:]*/i, "$1");
+
+	if (document.title.indexOf(url) < 0) {
+		var separatorPromise = browser.storage.local.get("separator");
+		separatorPromise.then(updateTitle);
+	}
 
-function getTitle(){
-    var url = document.URL;
-    
-    url = url.replace(/(https?:\/\/[^\/]*\/)[^:]*/i,"$1");
-    
-    if(document.title.indexOf(url) < 0){
-        return document.title + " | " + url;
-    } else {
-        return document.title;
-    }
+	function updateTitle(result) {
+		let separator = "|";
+		if (result.separator) {
+			separator = result.separator;
+		}
+		document.title = document.title + " " + separator + " " + url;
+	}
```

Comment:
I understand why the change is required by using the promise in that way. I would probably have used an options object that would be initialized  at page load and updated on storage change. This is to keep the updateUrl work the most simple as possible because some website updates their title quite often and we need to run this function on every title change.

Reply:
Ah, I hadn't though about that. Makes sense.